cobc/codegen.c

```diff
@@ -13302,11 +13302,20 @@ codegen (struct cb_program *prog, const char *translate_name)
 			break;
 		}
 		subsequent_call = 1;
-		if (current_program->flag_file_global
-		 && current_program->next_program->nested_level) {
-			has_global_file = 1;
+		/* set has_global_file if needed, only for sub-programs of this program */
+		if ( current_program->flag_file_global
+		     && current_program->next_program->nested_level >
```

Comment:
somehow the indent got mixed / lost; this would be the "GC default":

```c
		/* set has_global_file if needed, only for sub-programs of this program */
		if (current_program->flag_file_global
		 && current_program->next_program->nested_level >
		    current_program->nested_level) {
			if (!has_global_file) {
				has_global_file = 1 ;
				has_global_file_level = current_program->nested_level ;
			}
```

Reply:
should be ok now